src/components/clusters/Clusters/Clusters.test.tsx

```diff
@@ -0,0 +1,264 @@
+import React from 'react';
+
+import { screen, withState } from '~/testUtils';
+
+import { Clusters } from './Clusters';
+
+// Mock child components to simplify testing
+jest.mock('~/components/clusters/ClusterListMultiRegion', () => ({
+  __esModule: true,
+  default: () => <div data-testid="list-tab">Cluster List Content</div>,
+}));
+
+jest.mock('../ClusterTransfer/ClusterTransferList', () => ({
+  __esModule: true,
+  default: () => <div data-testid="cluster-transfer-list">Cluster Transfer List Content</div>,
+}));
+
+jest.mock('./ClustersPageHeader', () => ({
+  ClustersPageHeader: () => <div data-testid="clusters-page-header">Clusters Header</div>,
+}));
+
+const mockNavigate = jest.fn();
+const mockLocation = {
+  pathname: '/clusters',
+  search: '',
+  hash: '',
+  state: null,
+  key: 'default',
+};
+
+jest.mock('~/common/routing', () => ({
+  useNavigate: () => mockNavigate,
+}));
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useLocation: () => mockLocation,
+}));
+
+describe('<Clusters />', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+    mockLocation.hash = '';
+  });
+
+  describe('Initial Rendering', () => {
+    it('renders the clusters page with header and tabs', () => {
+      withState({}, true).render(<Clusters />);
+
+      expect(screen.getByTestId('clusters-page-header')).toBeInTheDocument();
+      expect(screen.getByRole('region', { name: 'Clusters' })).toBeInTheDocument();
+    });
+
+    it('renders both tab titles', () => {
+      withState({}, true).render(<Clusters />);
+
+      expect(screen.getByRole('tab', { name: 'Cluster List' })).toBeInTheDocument();
+      expect(screen.getByRole('tab', { name: 'Cluster Requests' })).toBeInTheDocument();
+    });
+
+    it('defaults to the "list" tab when no hash is present', () => {
+      mockLocation.hash = '';
```

Comment:
why do we need to reset `hash` again here if the 'before' function already takes care of that?

Reply:
Well, actually I copied the line around for other test cases and set the value in some of them.  They are not hurting anything but tried to remove them all anyway.